### PR TITLE
Refactor Full rank correlatin matrices, Euclidean, Graphspace, Hypersphere

### PR DESCRIPTION
**Pull request by SaitejaUtpala**

### CODE REVIEW

File: tests/tests_geomstats/test_euclidean.py [hunk 1]
```diff
@@ -1,373 +1,442 @@
...
+                ),
+                dict(dim=2, point=one_p, base_point=one_bp, expected=one_p - one_bp),
+                dict(dim=2, point=one_p, base_point=n_bps, expected=one_p - n_bps),
+                dict(dim=2, point=n_ps, base_point=one_bp, expected=n_ps - one_bp),
+                dict(dim=2, point=n_ps, base_point=n_bps, expected=n_ps - n_bps),
+            ]
+            return self.generate_tests(smoke_data)
+
+        def inner_product_data(self):
+            tangent_vec_1 = [[2.0, 1.0], [-2.0, -4.0], [-5.0, 1.0]]
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_euclidean.py:139]: this should be n_tangent_vecs_1

File: tests/tests_geomstats/test_euclidean.py [hunk 2]
```diff
@@ -1,373 +1,442 @@
...
+            tangent_vec_2 = [[2.0, 10.0], [8.0, -1.0], [-3.0, 6.0]]
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_euclidean.py:140]: this should be n_tangent_vecs2

File: tests/tests_geomstats/test_euclidean.py [hunk 3]
```diff
@@ -1,373 +1,442 @@
...
+                    dim=2, point=[2.0, 10.0], base_point=[0.0, 1.0], expected=[2.0, 9.0]
+                ),
+                dict(dim=2, point=one_p, base_point=one_bp, expected=one_p - one_bp),
+                dict(dim=2, point=one_p, base_point=n_bps, expected=one_p - n_bps),
+                dict(dim=2, point=n_ps, base_point=one_bp, expected=n_ps - one_bp),
+                dict(dim=2, point=n_ps, base_point=n_bps, expected=n_ps - n_bps),
+            ]
+            return self.generate_tests(smoke_data)
+
+        def inner_product_data(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_euclidean.py:138]: ok

File: tests/tests_geomstats/test_euclidean.py [hunk 4]
```diff
@@ -1,373 +1,442 @@
...
-
-        result = self.space.belongs(point)
-        self.assertTrue(result)
-
-    def test_random_point(self):
-        result = self.space.random_point()
-
-        self.assertAllClose(gs.shape(result), (self.dimension,))
-
-    def test_inner_product_matrix(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_euclidean.py:181]: ok

File: tests/tests_geomstats/test_euclidean.py [hunk 5]
```diff
@@ -1,373 +1,442 @@
...
-        result = self.metric.metric_matrix()
-
-        expected = gs.eye(self.dimension)
-
-        self.assertAllClose(result, expected)
-
-    def test_inner_product(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_euclidean.py:188]: ok

File: tests/tests_geomstats/test_euclidean.py [hunk 6]
```diff
@@ -1,373 +1,442 @@
...
-        point_a = gs.array([0.0, 1.0])
-        point_b = gs.array([2.0, 10.0])
-
-        result = self.metric.inner_product(point_a, point_b)
-        expected = 10.0
-
-        self.assertAllClose(result, expected)
-
-    def test_inner_product_vectorization_single_sample(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_euclidean.py:197]: ok

File: tests/tests_geomstats/test_euclidean.py [hunk 7]
```diff
@@ -1,373 +1,442 @@
...
-
-        one_point_a = gs.array([0.0, 1.0])
-        one_point_b = gs.array([2.0, 10.0])
-
-        result = self.metric.inner_product(one_point_a, one_point_b)
-        expected = 10.0
-        self.assertAllClose(gs.shape(result), ())
-        self.assertAllClose(result, expected)
-
-    def test_inner_product_vectorization_n_samples(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_euclidean.py:230]: ok

File: tests/tests_geomstats/test_euclidean.py [hunk 8]
```diff
@@ -1,373 +1,442 @@
...
-        expected = gs.array([10.0, -1.0, 6.0])
-        self.assertAllClose(gs.shape(result), (n_samples,))
-        self.assertAllClose(result, expected)
-
-        result = self.metric.inner_product(n_points_a, n_points_b)
-        expected = gs.array([14.0, -12.0, 21.0])
-        self.assertAllClose(gs.shape(result), (n_samples,))
-        self.assertAllClose(result, expected)
-
-    def test_squared_norm(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_euclidean.py:307]: ok

File: tests/tests_geomstats/test_euclidean.py [hunk 9]
```diff
@@ -1,373 +1,442 @@
...
-        point = gs.array([-2.0, 4.0])
-
-        result = self.metric.squared_norm(point)
-        expected = 20.0
-
-        self.assertAllClose(result, expected)
-
-    def test_norm(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_euclidean.py:315]: ok

File: tests/tests_geomstats/test_euclidean.py [hunk 10]
```diff
@@ -1,373 +1,442 @@
...
+        n_points_list = random.sample(range(1, 7), 5)
+        n_samples_list = random.sample(range(1, 7), 5)
+        n_points_a_list = random.sample(range(1, 7), 5)
+        n_points_b_list = [1]
+        batch_size_list = random.sample(range(2, 7), 5)
+        alpha_list = [1] * 5
+        n_rungs_list = [1] * 5
+        scheme_list = ["pole"] * 5
+
+        def exp_data(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_euclidean.py:91]: ok

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 11]
```diff
@@ -1,148 +1,237 @@
...
+from tests.parametrizers import LevelSetParametrizer, Parametrizer
+
+
+class TestFullRankCorrelationMatrices(TestCase, metaclass=LevelSetParametrizer):
+
+    space = FullRankCorrelationMatrices
+    skip_test_extrinsic_intrinsic_composition = True
+    skip_test_intrinsic_extrinsic_composition = True
+
+    class TestDataRankFullRankCorrelationMatrices(LevelSetTestData):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:25]: Ditto TestData should come at the end of the class name

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 12]
```diff
@@ -1,148 +1,237 @@
...
-    def setup_method(self):
-        gs.random.seed(12)
-        n = 3
-        self.n = n
-        bundle = CorrelationMatricesBundle(n)
-        self.bundle = bundle
-        self.base = bundle.base
-        self.quotient_metric = FullRankCorrelationAffineQuotientMetric(n)
-
-    def test_belongs(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:24]: ok

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 13]
```diff
@@ -1,148 +1,237 @@
...
+
+
+class TestCorrelationMatricesBundle(TestCase, metaclass=Parametrizer):
+    space = CorrelationMatricesBundle
+
+    class TestDataCorrelationMatricesBundle(TestData):
+        n_list = random.sample(range(2, 6), 3)
+        n_samples_list = random.sample(range(1, 7), 3)
+
+        def riemannian_submersion_belongs_to_base_data(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:67]: To add to design doc: this could be a property to test in a FiberBundleParametrizer

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 14]
```diff
@@ -1,148 +1,237 @@
...
-        self.bundle = bundle
-        self.base = bundle.base
-        self.quotient_metric = FullRankCorrelationAffineQuotientMetric(n)
-
-    def test_belongs(self):
-        point = self.base.random_point()
-        result = self.base.belongs(point)
-        self.assertTrue(result)
-
-    def test_riemannian_submersion(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:29]: ok

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 15]
```diff
@@ -1,148 +1,237 @@
...
+
+        def riemannian_submersion_belongs_to_base_data(self):
+            random_data = []
+            for n, n_samples in zip(self.n_list, self.n_samples_list):
+                bundle = CorrelationMatricesBundle(n)
+                point = bundle.base.random_point(n_samples)
+                random_data.append(dict(n=n, point=point))
+            return self.generate_tests([], random_data)
+
+        def lift_riemannian_submersion_composition_data(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:75]: ok

Comment (ninamiolane) [reply tests/tests_geomstats/test_full_rank_correlation_matrices.py:75]: To add to design doc: this could be a property to test in a FiberBundleParametrizer

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 16]
```diff
@@ -1,148 +1,237 @@
...
-        result = self.base.belongs(point)
-        self.assertTrue(result)
-
-    def test_riemannian_submersion(self):
-        mat = self.bundle.random_point()
-        point = self.bundle.riemannian_submersion(mat)
-        result = self.base.belongs(point)
-        self.assertTrue(result)
-
-    def test_lift_and_riemannian_submersion(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:35]: ok

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 17]
```diff
@@ -1,148 +1,237 @@
...
+
+        def lift_riemannian_submersion_composition_data(self):
+            random_data = []
+            for n, n_samples in zip(self.n_list, self.n_samples_list):
+                bundle = CorrelationMatricesBundle(n)
+                point = bundle.base.random_point(n_samples)
+                random_data.append(dict(n=n, point=point))
+            return self.generate_tests([], random_data)
+
+        def tangent_riemannian_submersion_data(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:83]: ok

To add to design doc: this could be a property to test in a FiberBundleParametrizer

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 18]
```diff
@@ -1,148 +1,237 @@
...
-        result = self.base.belongs(point)
-        self.assertTrue(result)
-
-    def test_lift_and_riemannian_submersion(self):
-        point = self.base.random_point()
-        mat = self.bundle.lift(point)
-        result = self.bundle.riemannian_submersion(mat)
-        self.assertAllClose(result, point)
-
-    def test_tangent_riemannian_submersion(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:41]: ok

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 19]
```diff
@@ -1,148 +1,237 @@
...
+            random_data = []
+            for n, n_samples in zip(self.n_list, self.n_samples_list):
+                bundle = CorrelationMatricesBundle(n)
+                mat = bundle.random_point()
+                point = bundle.riemannian_submersion(mat)
+                vec = bundle.random_point(n_samples)
+                random_data.append(dict(n=n, vec=vec, point=point))
+            return self.generate_tests([], random_data)
+
+        def vertical_projection_tangent_submersion_data(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:93]: ok
To add to design doc: this could be a property to test in a FiberBundleParametrizer

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 20]
```diff
@@ -1,148 +1,237 @@
...
-
-    def test_tangent_riemannian_submersion(self):
-        mat = self.bundle.random_point()
-        point = self.bundle.riemannian_submersion(mat)
-        vec = self.bundle.random_point()
-        tangent_vec = self.bundle.tangent_riemannian_submersion(vec, point)
-        result = self.base.is_tangent(tangent_vec, point)
-        self.assertTrue(result)
-
-    def test_vertical_projection(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:49]: ok

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 21]
```diff
@@ -1,148 +1,237 @@
...
+            random_data = []
+            for n in n_list:
+                bundle = CorrelationMatricesBundle(n)
+                point = bundle.random_point(2)
+                random_data.append(dict(n=n, point_a=point[0], point_b=point[1]))
+            return self.generate_tests([], random_data)
+
+    testing_data = TestDataCorrelationMatricesBundle()
+
+    def test_riemannian_submersion_belongs_to_base(self, n, point):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:153]: ok

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 22]
```diff
@@ -1,148 +1,237 @@
...
+        bundle = self.space(n)
+        result = bundle.base.belongs(bundle.riemannian_submersion(gs.array(point)))
+        self.assertAllClose(gs.all(result), gs.array(True))
+
+    def test_lift_riemannian_submersion_composition(self, n, point):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:158]: ok

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 23]
```diff
@@ -1,148 +1,237 @@
...
+        bundle = self.space(n)
+        result = bundle.riemannian_submersion(bundle.lift(gs.array(point)))
+        self.assertAllClose(result, gs.array(point))
+
+    def test_tangent_riemannian_submersion(self, n, vec, point):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:163]: ok

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 24]
```diff
@@ -1,148 +1,237 @@
...
+        bundle = self.space(n)
+        tangent_vec = bundle.tangent_riemannian_submersion(
+            gs.array(vec), gs.array(point)
+        )
+        result = gs.all(bundle.is_tangent(gs.array(tangent_vec), gs.array(point)))
+        self.assertAllClose(result, gs.array(True))
+
+    def test_vertical_projection_tangent_submersion(self, n, vec, mat):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:171]: ok

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 25]
```diff
@@ -1,148 +1,237 @@
...
+        result = bundle.tangent_riemannian_submersion(proj, gs.array(mat))
         expected = gs.zeros_like(vec)
-        self.assertAllClose(result, expected)
+        self.assertAllClose(result, gs.array(expected))
 
-    def test_horizontal_projection(self):
-        mat = self.bundle.random_point()
-        vec = self.bundle.random_point()
-        horizontal_vec = self.bundle.horizontal_projection(vec, mat)
+    def test_horizontal_projection(self, n, vec, mat):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:179]: ok

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 26]
```diff
@@ -1,148 +1,237 @@
...
+        def vertical_projection_tangent_submersion_data(self):
+            random_data = []
+            for n in self.n_list:
+                bundle = CorrelationMatricesBundle(n)
+                mat = bundle.random_point(2)
+                vec = SymmetricMatrices(n).random_point(2)
+                random_data.append(dict(n=n, vec=vec, mat=mat))
+            return self.generate_tests([], random_data)
+
+        def horizontal_projection_data(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:102]: ok

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 27]
```diff
@@ -1,148 +1,237 @@
...
-
-    def test_horizontal_lift_and_tangent_riemannian_submersion(self):
-        mat = self.base.random_point()
-        vec = self.base.random_point()
-        tangent_vec = self.base.to_tangent(vec, mat)
-        horizontal = self.bundle.horizontal_lift(tangent_vec, base_point=mat)
-        result = self.bundle.tangent_riemannian_submersion(horizontal, mat)
+        self.assertAllClose(is_horizontal, gs.array(True))
+
+    def test_horizontal_lift_is_horizontal(self, n, tangent_vec, mat):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:190]: ok

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 28]
```diff
@@ -1,148 +1,237 @@
...
+        bundle = self.space(n)
+        lift = bundle.horizontal_lift(gs.array(tangent_vec), gs.array(mat))
+        result = gs.all(bundle.is_horizontal(lift, gs.array(mat)))
+        self.assertAllClose(result, gs.array(True))
+
+    def test_vertical_projection_is_vertical(self, n, tangent_vec, mat):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:196]: ok

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 29]
```diff
@@ -1,148 +1,237 @@
...
+    def test_vertical_projection_tangent_submersion(self, n, vec, mat):
+        bundle = self.space(n)
+        tangent_vec = bundle.to_tangent(vec, mat)
+        proj = bundle.vertical_projection(gs.array(tangent_vec), gs.array(mat))
+        result = bundle.tangent_riemannian_submersion(proj, gs.array(mat))
         expected = gs.zeros_like(vec)
-        self.assertAllClose(result, expected)
+        self.assertAllClose(result, gs.array(expected))
 
-    def test_horizontal_projection(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:59]: ok

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 30]
```diff
@@ -1,148 +1,237 @@
...
         product_1 = Matrices.mul(horizontal_vec, inverse)
         product_2 = Matrices.mul(inverse, horizontal_vec)
-        is_horizontal = self.base.is_tangent(
-            product_1 + product_2, mat, atol=gs.atol * 10
+        is_horizontal = gs.all(
+            bundle.base.is_tangent(product_1 + product_2, mat, atol=gs.atol * 10)
         )
-        self.assertTrue(is_horizontal)
-
-    def test_horizontal_lift_and_tangent_riemannian_submersion(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:71]: ok

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 31]
```diff
@@ -1,148 +1,237 @@
...
+
+    def test_horizontal_lift_and_tangent_riemannian_submersion(
+        self, n, tangent_vec, mat
+    ):
+        bundle = self.space(n)
+        horizontal = bundle.horizontal_lift(gs.array(tangent_vec), gs.array(mat))
+        result = bundle.tangent_riemannian_submersion(horizontal, gs.array(mat))
         self.assertAllClose(result, tangent_vec)
 
-    def test_is_horizontal(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:79]: ok

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 32]
```diff
@@ -1,148 +1,237 @@
...
-        mat = self.base.random_point()
-        vec = self.base.random_point()
-        tangent_vec = self.base.to_tangent(vec, mat)
-        horizontal = self.bundle.horizontal_lift(tangent_vec, mat)
-        result = self.bundle.is_horizontal(horizontal, mat)
-        self.assertTrue(result)
-
-    def test_is_vertical(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:87]: ok

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 33]
```diff
@@ -1,148 +1,237 @@
...
+        self.assertAllClose(result, gs.array(True))
+
+    def test_vertical_projection_is_vertical(self, n, tangent_vec, mat):
+        bundle = self.space(n)
+        proj = bundle.vertical_projection(gs.array(tangent_vec), gs.array(mat))
+        result = gs.all(bundle.is_vertical(proj, gs.array(mat)))
+        self.assertAllClose(result, gs.array(True))
+
+    @autograd_tf_and_torch_only
+    def test_align_is_horizontal(self, n, point_a, point_b):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:203]: maybe: log_after_align_is_horizontal?

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 34]
```diff
@@ -1,148 +1,237 @@
...
+        bundle = self.space(n)
+        aligned = bundle.align(point_a, point_b, tol=1e-10)
+        log = bundle.ambient_metric.log(aligned, point_b)
+        result = bundle.is_horizontal(log, point_b, atol=gs.atol * 10000)
+        self.assertAllClose(result, gs.array(True))
+
+    def test_horizontal_lift_and_tangent_riemannian_submersion(
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:210]: ok

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 35]
```diff
@@ -1,148 +1,237 @@
...
-        def inner_prod(tangent_vec_a, tangent_vec_b, base_point):
-            affine_part = self.bundle.ambient_metric.inner_product(
-                tangent_vec_a, tangent_vec_b, base_point
-            )
-            n = tangent_vec_b.shape[-1]
 
-            inverse_base_point = GeneralLinear.inverse(base_point)
-            operator = gs.eye(n) + base_point * inverse_base_point
-            inverse_operator = GeneralLinear.inverse(operator)
+class TestFullRankCorrelationAffineQuotientMetric(TestCase, metaclass=Parametrizer):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:219]: why not RiemannianMetricParametrizer?

Comment (ninamiolane) [reply tests/tests_geomstats/test_full_rank_correlation_matrices.py:219]: Note for improvements: ultimately, it should have QuotientMetricParametrizer as metaclass

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 36]
```diff
@@ -1,148 +1,237 @@
...
-        self.assertTrue(result)
-
-    def test_is_vertical(self):
-        mat = self.bundle.random_point()
-        tangent_vec = self.bundle.random_point()
-        vertical = self.bundle.vertical_projection(tangent_vec, mat)
-        result = self.bundle.is_vertical(vertical, mat)
-        self.assertTrue(result)
-
-    def test_inner_product(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:94]: this is missing

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py [hunk 37]
```diff
@@ -1,148 +1,237 @@
...
-        mat = self.base.random_point()
-        vecs = self.bundle.random_point(2)
-        tangent_vecs = self.base.to_tangent(vecs, mat)
-        result = self.quotient_metric.inner_product(
-            tangent_vecs[0], tangent_vecs[1], base_point=mat
-        )
-        expected = inner_prod(tangent_vecs[0], tangent_vecs[1], base_point=mat)
-        self.assertAllClose(result, expected)
-
-    def test_exp_and_belongs(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_full_rank_correlation_matrices.py:124]: this is missing

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py (hunk 35 above)
Comment (SaitejaUtpala) [reply tests/tests_geomstats/test_full_rank_correlation_matrices.py:219]: Tests are failing. (I explained this in standup texts). For time being this is being skipped

File: tests/data_generation.py [hunk 38]
```diff
@@ -617,6 +638,7 @@ def _log_exp_composition_data(
         connection_args_list,
         space_list,
         n_samples_list,
+        smoke_data=None,
```
[the remark this answers is not in the document]
Comment (SaitejaUtpala) [reply tests/data_generation.py:641]: this because. there are tests in `hypersphere`  that wants to test log_exp composition in some edge cases. It's better that we don't have to write another another duplication test for this I thought.

File: tests/tests_geomstats/test_full_rank_correlation_matrices.py (hunk 36 above)
Comment (SaitejaUtpala) [reply tests/tests_geomstats/test_full_rank_correlation_matrices.py:94]: yeah. I removed.  this is not ideal way of testing. this is making sure two different implementations of innerproduct match. what if the test_implementation is wrong.

File: tests/tests_geomstats/test_euclidean.py [hunk 39]
```diff
@@ -1,373 +1,442 @@
...
-        self.n_points_b = gs.array([[2.0, 10.0], [8.0, -1.0], [-3.0, 6.0]])
-
-    def test_random_point_and_belongs(self):
-        point = self.space.random_point()
-        result = self.space.belongs(point)
-        expected = True
-
-        self.assertAllClose(result, expected)
-
-    def test_is_tangent(self):
```
[the remark this answers is not in the document]
Comment (SaitejaUtpala) [reply tests/tests_geomstats/test_euclidean.py:30]: yeah.. this will be in the next improvements. I removed it because, we will be adding again. Also I am thinking to_tangent is identity for only points on manifold. not necessarily for every point because. If we consider symmetric matrices, not very matrix is symmetric matrix.

[the remark this answers is not in the document]
Comment (SaitejaUtpala) [reply tests/tests_geomstats/test_euclidean.py:30]: I removed it because, I didn't wanna spend on time on something I will delete it and to_tangent, is_tangent is already being tested in to_tangent_is_tangent way. so codecov is not being changed.

[the remark this answers is not in the document]
Comment (ninamiolane) [reply tests/tests_geomstats/test_euclidean.py:30]: ok

**Review by ninamiolane — APPROVED** [review]
Good, many thanks!!
